python/calibration.py

```diff
@@ -2,11 +2,12 @@
 import yaml
 from os import chdir
 from pathlib import Path
-from ngen.cal.configuration import General, Model
-from ngen.cal.meta import CalibrationMeta
-from ngen.cal.search import dds, dds_set
+from ngen.cal.configuration import General
+from ngen.cal.search import dds, dds_set, pso_search
+from ngen.cal.strategy import Algorithm
+from ngen.cal.agent import Agent
 
-def main(general: General, model: Model):
+def main(general: General, model_conf):
```

Comment:
Wellllll, you recon you could add that new type hint here?

Reply:
As, so the `__main__` no longer builds all the data models.  It validates the `General` configuration block in the input file, and passes the rest of the loaded yaml dict to `main` which uses it to initialize the `Agent` model.  I can add a type hint here, but it will be pretty generic, e.g.
```python
def main(general: General, model_conf: Mapping[str, Any]):
```

Reply:
a9783ad65eca119a7c7e0b2a7934e49891367222

Reply:
a9783ad65eca119a7c7e0b2a7934e49891367222

Reply:
Hmm....after digging into this more, `model_conf` is essentially a `Model` base type still...but I think because I'm not creating the explicit `Model` object until the `Agent` is created, I decided to not use that as the type hint, so it is essentially a `TypedDict` like you mention below, but to specifiy that type, I think I would be redefining the `Model` just as a dict and not a pydantic data model 🤔

Reply:
It is a shame that you can't do `model_conf: Model.dict` or something like `model_conf: Dict[keyof Model, typeof keyof Model]`. I think what you proposed above makes the most sense.